### PR TITLE
HVG-201 Feature/renewal home

### PR DESCRIPTION
**Pull request by MathiasOhrstrom**
Item from design: [home_tab_renewal_card]

### Checklist

- [x] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt [hunk 1]
```diff
@@ -230,6 +233,26 @@ class HomeAdapter(
...
+                fragmentManager: FragmentManager,
+                retry: () -> Unit,
+                requestBuilder: RequestBuilder<PictureDrawable>,
+                tracker: HomeTracker,
+                marketProvider: MarketProvider
+            ): Any? = with(binding) {
+                if (data !is HomeModel.UpcomingRenewal) {
+                    return invalid(data)
+                }
+                root.adapter = UpcomingRenewalAdapter(tracker).also {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt:250]: We should not have a nested `RecyclerView` here. Instead, directly generate one card to be added to this `RecyclerView`.

File: app/src/main/res/drawable/ic_document.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (hedvigoscar) [inline app/src/main/res/drawable/ic_document.xml:1]: Is this file `avocado`ified`? It looks kind of weird 😬

File: testdata/src/main/java/com/hedvig/app/testdata/feature/home/builders/HomeDataBuilder.kt [hunk 3]
```diff
@@ -16,7 +16,8 @@ data class HomeDataBuilder(
         CommonClaimBuilder(title = "Trasig telefon").build(),
         CommonClaimBuilder(title = "Försenat bagage").build()
     ),
-    private val importantMessages: List<HomeQuery.ImportantMessage> = emptyList()
+    private val importantMessages: List<HomeQuery.ImportantMessage> = emptyList(),
+    private val upcomingRenewal: HomeQuery.UpcomingRenewal? = null
```
Comment (hedvigoscar) [inline testdata/src/main/java/com/hedvig/app/testdata/feature/home/builders/HomeDataBuilder.kt:20]: I think we can simplify this to something like `renewalDate: LocalDate? = null` and then generate the entire `HomeQuery.UpcomingRenewal` if it is not null. That saves effort on the calling side.

**Review by hedvigoscar — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt (hunk 1 above)
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt:250]: What is the gain by doing it that way?

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt:250]: One `RecyclerView` will mean that we actually recycle properly.

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt [hunk 4]
```diff
@@ -82,6 +86,7 @@ class HomeAdapter(
         HomeModel.Error -> R.layout.home_error
         is HomeModel.PSA -> R.layout.home_psa
         is HowClaimsWork -> R.layout.how_claims_work_button
+        is HomeModel.UpcomingRenewal -> R.layout.upconing_renewal_card
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt:89]:
```suggestion
        is HomeModel.UpcomingRenewal -> R.layout.upcoming_renewal_card
```

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt [hunk 5]
```diff
@@ -181,6 +188,10 @@ class HomeFragment : Fragment(R.layout.home_fragment) {
         .filterNotNull()
         .map { HomeModel.PSA(it) }
 
+    private fun upcomingRenewals(contracts: List<HomeQuery.Contract>) =
+        contracts.filter { it.upcomingRenewal != null }
+            .map { HomeModel.UpcomingRenewal(it) }
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt:193]:
```suggestion
        contracts.mapNotNull { c -> c.upcomingRenewal?.let { HomeModel.UpcomingRenewal(it) } }
```

This is an alternative form you could use, it's by no means required though 🙂

**Review by hedvigoscar — APPROVED** [review]
Approved upon fixing comments!

**Pull request merged**